sequgen/deterministic/boxcar.py

```diff
@@ -0,0 +1,30 @@
+import numpy
+
+
+def boxcar(t_predict, location=1.0, height=1.0, width=1.0):
+    """Generate a time series containing boxcar function.
+
+    Args:
+      t_predict (Numpy array):
+        Where you want the model to generate predictions.
+
+      location (float):
+        The start (left point) of the plateau.
+
+      height (float):
+        The height of the plateau.
+
+      width (float):
+        The width of the plateau.
+
+    Returns:
+      Numpy array of shape equal to t_predict containing the signal with the boxcar plateau.
+    """
+
+    lower = min(location, location+width)
```

Comment:
this is a bit of a surprise, are you expecting negative widths? I would prefer to assert that width > 0

Reply:
Ok, negative widths have been blocked by an assert and (suggestion from Stefan) the parameters `location` and `width` have become required

Reply:
good stuff